text/todo-generic-payload.md

```diff
@@ -0,0 +1,368 @@
+
+**Extensions:**
+
+| Field Name  | Type        | Description                                                                          |
+| ----------- | ----------- | ------------------------------------------------------------------------------------ |
+| power_draw  | uint16_t    | Power consumption of the payload in mW. 0 if not available.                          |
+| temperature | uint16_t    | Temperature of the payload (0.01 °C per unit, 0 = 0°C). UINT16_MAX if not available. |
+| mass        | uint16_t    | Mass of the payload in grams. 0 if not available.                                    |
+| torque_arm  | uint16_t[3] | Torque arm in Payload Frame (mm)                                                     |
+
+#### GENERIC_PAYLOAD_FUNCTION_STATUS (Message ID: 60001)
+
+Generic payload function status.
+
+| Field Name    | Type       | Description                                                                      |
+| ------------- | ---------- | -------------------------------------------------------------------------------- |
+| index         | uint16_t   | Index of this function on the generic payload.                                   |
+| type          | uint8_t    | Type of function. See GENERIC_PAYLOAD_FUNCTION_TYPE.                             |
+| value_type    | uint8_t    | Data type for value/min/max fields. See GENERIC_PAYLOAD_VALUE_TYPE.              |
+| enabled       | uint8_t    | 0: Disabled, 1: Enabled                                                          |
+| value_low     | uint8_t[4] | Lower 32 bits of current value. For 32-bit types, this is the entire value.      |
+| min_low       | uint8_t[4] | Lower 32 bits of minimum value (for CONTINUOUS/DISCRETE types).                  |
+| max_low       | uint8_t[4] | Lower 32 bits of maximum value (for CONTINUOUS/DISCRETE types).                  |
+| control_modes | uint16_t   | Bitmask of supported control modes. See GENERIC_PAYLOAD_CONTROL_MODE_FLAGS.      |
+| timeout_ms    | uint32_t   | Timeout in milliseconds for MOMENTARY mode. 0 means use default timeout (100ms). |
+| name          | char[32]   | Name of the function for UI display. NULL-terminated string.                     |
+| units         | char[16]   | Units for the value field (optional). NULL-terminated string.                    |
+
+**Extensions:**
+
+| Field Name | Type       | Description                                                 |
+| ---------- | ---------- | ----------------------------------------------------------- |
+| value_high | uint8_t[4] | Upper 32 bits of current value. Only used for 64-bit types. |
+| min_high   | uint8_t[4] | Upper 32 bits of minimum value. Only used for 64-bit types. |
+| max_high   | uint8_t[4] | Upper 32 bits of maximum value. Only used for 64-bit types. |
+
+#### GENERIC_PAYLOAD_FUNCTION_CONTROL (Message ID: 60002)
+
+Command to control a generic payload function.
+
```

Comment:
If we have a payload id we will need to have that in here too. Obviously you don't need it if we were considering purely mavlink payloads.